src/libraries/System.Private.CoreLib/src/System/IO/FileSystem.Windows.cs

```diff
@@ -182,7 +182,7 @@ public static void MoveFile(string sourceFullPath, string destFullPath, bool ove
             }
         }
 
-        private static SafeFileHandle OpenHandle(string fullPath, bool asDirectory)
+        private static SafeFileHandle OpenHandleToWriteAttributes(string fullPath, bool asDirectory)
         {
             string root = fullPath.Substring(0, PathInternal.GetRootLength(fullPath.AsSpan()));
```

Comment:
We shouldn't need to use Substring here.  Couldn't these two lines instead be something like:
```C#
if (fullPath.Length == PathInternal.GetRootLength(fullPath) && fullPath[1] == Path.VolumeSeparatorChar)
```
?